### PR TITLE
PIN-3367 [BE] Aggiunta campi gestione titolo gratuito purpose-process

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
Linked to [PIN-3366](https://github.com/pagopa/interop-be-purpose-management/pull/200)

[PIN-3366]: https://pagopa.atlassian.net/browse/PIN-3366?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -1061,11 +1066,17 @@ components:
           type: string
           minLength: 10
           maxLength: 250
+        isFreeOfCharge:
+          type: boolean
+          default: true
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: do we need a default?
this is an administrative declaration, shouldn't this be totally in charge of the user?

Comment (galales) [reply src/main/resources/interface-specification.yml]: this question is more for @beetlecrunch

Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml]: Let's remove this default value.

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 2]
```diff
@@ -100,10 +100,11 @@ final case class PurposeApiServiceImpl(
     val result: Future[Purpose] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       _              <- assertOrganizationIsAConsumer(organizationId, seed.consumerId)
-      tenant         <- tenantManagementService.getTenant(organizationId)
-      tenantKind     <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
-      clientSeed     <- seed.apiToDependency(schemaOnlyValidation = true)(tenantKind).toFuture
-      agreements     <- agreementManagementService.getAgreements(
+      _      <- Future.failed(MissingFreeOfChargeReason) whenA (seed.isFreeOfCharge && seed.freeOfChargeReason.isEmpty)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: don't use infix notation please
also, `whenA` is a cats method, and as such should not be used with Futures

**Review by TonioGela — CHANGES_REQUESTED** [review]
Replace that infix method and it's ok

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: Martin Odersky loves infix notation

**Review by TonioGela — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**